api/app/src/main/kotlin/packit/model/Role.kt

```diff
@@ -35,11 +35,22 @@ class Role(
         result = 31 * result + isUsername.hashCode()
         return result
     }
+
+    fun isHidden(): Boolean = isUsername
 }
 
 fun Role.toDto() =
     RoleDto(
-        name, rolePermissions.map { it.toDto() }, users.map { it.toBasicDto() }, id!!, isUsername
+        name,
+        rolePermissions.map { it.toDto() }.sortedByDescending {
+            it.tag == null && it.packet == null && it.packetGroup == null
```

Comment:
I copied this from `getSortedRoleDtos`. I'm not sure I really understand what it is trying to achieve tbh. Does it put global permissions at the top, and scoped ones further down?